Tools/clinic/clinic.py

```diff
@@ -2460,10 +2473,10 @@ def methoddef_flags(self):
             flags.append('METH_COEXIST')
         return '|'.join(flags)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return '<clinic.Function ' + self.name + '>'
 
-    def copy(self, **overrides):
+    def copy(self, **overrides) -> "Function":
```

Comment:
If you like, we can avoid the need for putting quotes around some forward references by adding `from __future__ import annotations` to the top of the file. That will only work for parameter annotations, attribute annotations, and return annotations, however. You'll still need quotes if you want to make a type alias to a forward reference, even if you have `from __future__ import annotations` enabled.

Reply:
I'm fine with the quotes. What's the preference in the typing community?

Reply:
Most heavily typed codebases use `from __future__ import annotations` because:
1. Most people find the quotes around forward references ugly
2. On older versions of Python, there's some hacks you can use involving `from __future__ import annotations` that let you use new syntax "before it's been released".
3. On codebases with a lot of type annotations, you can get a little bit of a speedup, as all your annotations are just strings, which means all those subscriptions in your annotations (`dict[str, Any]` etc.) don't need to be resolved.

Points (2) and (3) don't really apply here -- we don't need any typing features that were introduced after Python 3.10, and there's no need to micro-optimise. So that leaves only point (1), which is a purely subjective, aesthetic judgement call :)

Reply:
Ok, so I'm fine with the quotes :)